xtrack/line.py

```diff
@@ -1059,9 +1249,24 @@ def merge_consecutive_multipoles(self, inplace=True):
             else:
                 newline.append_element(ee, nn)
 
-        self.element_dict.update(newline.element_dict)
-        self.element_names = newline.element_names
-        return self
+        _lref = None
+        if self._var_management is not None:
+            import xdeps as xd
+            # Update the lref to point to the new element_dict
+            manager = xd.Manager()
+            _lref = manager.ref(newline.element_dict, 'element_refs')
+
+        if inplace:
+            self.element_names = newline.element_names
+            self.element_dict  = newline.element_dict
+            if _lref is not None:
+                self._var_management['lref'] = _lref
+            return self
+        elif _lref is not None:
+            # copy the var management
+            newline._init_var_management(dct=self._var_management_to_dict())
+            newline._var_management['lref'] = _lref
```

Comment:
Where are the expressions of the multipole strengths combined?
What happened if the two multipoles have different expressions for hxl and hyl?

Reply:
Adapted; line simplifications are no longer compatible with deferred expressions.